### ISSUE TITLE
Rename weave-feedstock

### ISSUE THREAD

**Issue by mstimberg** [conda-forge/admin-requests#766]
### Comment:

A while ago, the `weave` package was split-off from `scipy.weave` in maintenance-only mode (in particular, it was Python2-only with no ongoing development). Recently, the package has again seen a bit of development, in particular it has now experimental support for Python 3. More importantly, it has been renamed to `scipy-weave`, to release the name `weave` on pypi. For more details, see https://mail.python.org/archives/list/scipy-dev@python.org/message/AGFIP5TMTIYDZAF6B5WAFYCXJ66LAHDW/

The conda-forge recipe for `weave` is still referring to `weave` on pypi which is now an unrelated package. I guess the best approach would be to rename the `weave` package to `scipy-weave` on conda-forge as well? I don't know what the correct procedure is in this case – I can open a new PR on staged recipes for `scipy-weave`, but I don't know how to delete/deprecate an existing package. Otherwise I could of course update the package recipe so that it refers to `scipy-weave` on pypi, but I feel that would only lead to more confusion in the future.

Thanks for your help!

[EDIT: forgot a few links]
Current weave feedstock: https://github.com/conda-forge/weave-feedstock
Scipy-weave on Pypi (formerly known as weave): https://pypi.org/project/scipy-weave/
PR that renamed weave to scipy-weave: https://github.com/scipy/weave/pull/18
New, unrelated weave package on Pypi: https://pypi.org/project/weave/

**Comment by jaimergp** [comment]
Hey @mstimberg, thanks for the question!

There are two things to note here:

- Renaming feedstocks is not supported, so we'll have to archive the existing one after obtaining a new one through `staged-recipes`.
- I think it would be tricky to change what `weave` means. Many problems come to mind like conflicting versioning, existing environments with "the other weave" all of a sudden getting something different, etc. The safest option would be to publish the unrelated `weave` with a different name in conda-forge.

Alternatively, you can keep the `weave-feedstock` and change your recipe into a multi-output recipe where you publish `scipy-weave` and `weave` simply aliases to it. I think that's the less confusing story here. The Weight & Biases team can choose to publish their `weave` with a different name if required (e.g. `wandb-weave` or something).

**Comment by ocefpaf** [comment]
> Alternatively, you can keep the `weave-feedstock` and change your recipe into a multi-output recipe where you publish `scipy-weave` and `weave` simply aliases to it.

If we want to keep the namespace `weave` for `scipy-weave` this is the best solution. If we don't care about that we must archive this one to prevent it from becoming the "new weave," which is a different package that clobbers this namespace.

Namespace in conda-forge is first come first served. So the name `weave` belongs to "scipy-weave" in my view, to archive it or keep it with multiple outputs along side `scipy-weave`. You choice @conda-forge/weave team.

**Comment by mstimberg** [comment]
Thank you for your comments/suggestions @jaimergp and @ocefpaf. Just to be clear, I don't have a stake in this and I am not even involved with the development of `scipy-weave` in the first place – I only had a personal use case for it and needed it to be on `conda-forge`, so I created the recipe and maintained it. I am happy with any solution really, and I wouldn't feel particularly sorry for Weight & Biases if they cannot claim the `weave` name on `conda-forge`. I am just CC'ing @rgommers here, since he's maintaining `scipy-weave` (and was involved in the negotiations around releasing the pypi name).

Just to give a quick summary, it seems there are two solutions:
1. Archive `weave-feedstock` and go through the usual route for a new `scipy-weave-feedstock`
2. Turn `weave-feedstock` into a multi-output recipe, creating both `scipy-weave` and `weave` packages.

I'm happy with either solution, really, the first would be a bit easier for me, but slightly less convenient for existing users of `weave`. Although I am not sure whether this is really much of an issue: the conda-forge `weave` package is still Python2-only and the new `scipy-weave` package is obviously now Python3-only.

**Comment by rgommers** [comment]
Hey all. For context: the SciPy team decided to hand over the `weave` name on PyPI to Weights & Biases (and W&B supported SciPy development with a significant donation, so it was a win-win situation). During that rename to `scipy-weave` I took care of incorporating open PRs and making some maintenance updates for the package, most importantly Python 3.x support. That was a one-off effort, the package is effectively stagnant since there are much better alternatives now (Cython, Pythran, Numba - take your pick).

I'd be happy with whatever solution is easiest here, and also with repurposing the existing `weave` name for W&B if they would like to have it at some point. The download count for the last weave release is very small, this won't affect many users and we've had no complaints about the rename on PyPI.

**Comment by mstimberg** [comment]
Thanks @rgommers ! Ok, from my side then the preferred solution would be to close down the current `weave-feedstock` (i.e., archive it), and I will go through the usual staged recipes process for `scipy-weave`. If Weights&Biases will show up some day to get  the "new weave" onto conda-forge, I'll let them figure that out with conda-forge admins. Thanks everyone for your help!

PS: I don't think there's anything I can do myself to close down/archive `weave-feedstock`?

**Comment by jaimergp** [comment]
> PS: I don't think there's anything I can do myself to close down/archive `weave-feedstock`?

You need to open a PR in `admin-requests` (here) to request the archival. Create an issue in the feedstock too (maybe pointing to this discussion) to have a reference on why it was archived. Thanks!

**Comment by mstimberg** [comment]
After having had a closer look at [scipy/weave](https://github.com/scipy/weave/), I finally decided to not go ahead with submitting a new conda-forge package for `scipy-weave`. As @rgommers explained earlier, the package is not developed any further, and it will not be installable on up-to-date systems very soon. As it is, it is not installable with Python≥3.11, and its test suite does not work with numpy ≥ 1.25. Again, as @rgommers said, nowadays there are better alternatives, so it is unlikely that someone will invest the time for a major update. If someone does, by all means get in touch with me if you want me to maintain a new conda-forge package :wink: 

I have opened a new issue to archive the feedstock (admin-requests/issues/776), and have also opened an issue on the `weave-feedstock` to explain the situation to anyone stopping by (conda-forge/weave-feedstock#21).
